pyhealth/datasets/base_dataset.py

```diff
@@ -86,10 +85,17 @@ def __init__(
             self.__class__.__name__ if dataset_name is None else dataset_name
         )
         self.root = root
-        self.tables = tables
         self.code_mapping = code_mapping
         self.dev = dev
 
+        # if we are using a premade dataset, no basic tables need to be provided. 
+        if self.dataset_name in DATASET_BASIC_TABLES and [ table for table in tables if table in DATASET_BASIC_TABLES[self.dataset_name]]:
+            raise AttributeError(
+                f"Basic tables are parsed by default and do not need to be explicitly selected. Basic tables for {self.dataset_name}: {DATASET_BASIC_TABLES[self.dataset_name]}"
+            )
```

Comment:
Prior to these changes, whenever a user passes in a table which is included in the basic_tables dataset, you get this error, since PATIENTS doesn't have a dedicated parser and is parsed in parse_basic_info.
<img width="926" alt="Screenshot 2023-02-06 at 3 45 42 PM" src="https://user-images.githubusercontent.com/24256419/217406482-c7113c25-42ba-414f-aa55-2bef6d426f9f.png">

The handled error im proposing is to let the user know that they don't need to parse basic tables. I haven't implemented for other datasets yet, but plan to. 
<img width="926" alt="Screenshot 2023-02-06 at 4 03 38 PM" src="https://user-images.githubusercontent.com/24256419/217407003-1b76d16c-d084-45ee-aa22-12448bf2b8a7.png">